demoextendsymfonyform/src/Controller/DemoSupplierController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2020 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0).
+ * It is also available through the world-wide-web at this URL: https://opensource.org/licenses/AFL-3.0
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoExtendSymfonyForm\Controller;
+
+use PrestaShop\Module\DemoExtendSymfonyForm\Uploader\SupplierExtraImageUploader;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CannotDeleteImageException;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Exception\SupplierException;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+class DemoSupplierController extends FrameworkBundleAdminController
+{
+
+    /**
+     * Deletes image.
```

Comment:
A question we might keep for later: what about security ?

Reply:
adding     ```
 *
     * @AdminSecurity(
     *     "is_granted(['update'], request.get('_legacy_controller'))",
     *     message="You do not have permission to edit this.",
     *     redirectRoute="admin_suppliers_edit",
     *     redirectQueryParamsToKeep={"supplierId"}
     * )

gitves strange error: `The slug ROLE_MOD_TAB__UPDATE is invalid`
```

Reply:
Valid route should be: ROLE_MOD_TAB_ADMINCATEGORIES_UPDATE

Reply:
No `_legacy_controller` was defined => security system cannot guess the tab rights to use

We must provide a real Controller name, not `request.get('_legacy_controller')`
Example
```
 "is_granted(['update'], request.get('MyController'))",
```

Reply:
Tried it both SupplierController and `AdminSuppliers`, something more is missing.

Reply:
OK I'll check it